backend/server.js

```diff
@@ -0,0 +1,226 @@
+  address: String
+})
+
+/**
+ * From https://github.com/BrightID/BrightID/blob/9a9570dd0b1a39aafa770e1f3bf745259b023bc5/BrightID/src/utils/encoding.js#L37
+ */
+function b64ToUrlSafeB64 (s) {
+  const alts = {
+    '/': '_',
+    '+': '-',
+    '=': '',
+  };
+  return s.replace(/[/+=]/g, (c) => alts[c]);
+}
+
+function hash (data)  {
+  const h = CryptoJS.SHA256(data);
+  const b = h.toString(CryptoJS.enc.Base64);
+  return b64ToUrlSafeB64(b);
+};
+
+const User = mongoose.model('User', userSchema)
+
+/**
+ * Be assigned a newly generated UUID for the BrightID
+ * for your crypto wallet address
+ * @param address your wallet address
+ */
+app.post('/deep-link/:address',async (req, res) => {
+  // If user already assigned with address, then don't bother
+  User.findOne({address: req.params.address}, (err, user) => {
+    if (user) {
+      const deepLink = 'brightid://link-verification/' +
+        encodeURIComponent(BRIGHTID_NODE_DOMAIN) + `/${CONTEXT}/` + user.contextId
+      res.send(deepLink)
+    } else {
+      const contextId = uuidv4()
+      const newUser = new User({ 
+        contextId, verified: false, 
+        creationDate: Date.now(), address: req.params.address
+      })
+      newUser.save((err) => {
+        if (err) {
+          res.status(500).send('Server Error')
+        } else {
+          const deepLink = 'brightid://link-verification/' +
+            BRIGHTID_NODE_DOMAIN + '/DollarForEveryone/' + contextId
+          res.send(deepLink)
+        }
+      })
+    }
+  })
+})
+
+/**
+ * From BrightID for Applications
+ */
+function getMessage(op) {
+  const signedOp = {};
+  for (let k in op) {
+    if (['sig', 'sig1', 'sig2', 'hash'].includes(k)) {
+      continue;
+    }
+    signedOp[k] = op[k];
+  }
+  return stringify(signedOp);
+ }
+
+/**
+ * From BrightID for Applications
+ */
+function strToUint8Array(str) {
+  return new Uint8Array(Buffer.from(str, 'ascii'));
+}
+
+/**
+ * From BrightID for Applications
+ */
+function uInt8ArrayToB64(array) {
+  const b = Buffer.from(array);
+  return b.toString('base64');
+}
+
+
+const sponsorUser = async (contextId) => {
+  
+  const sk = nacl.util.decodeBase64(process.env.CONTEXT_SK)
+  
+  // From BrightID for Applications.
+  const timestamp = Date.now()
+  const op = {
+    app: CONTEXT,
+    contextId,
+    name: 'Sponsor',
+    timestamp,
+    v: 5,
+  }
+  const message = getMessage(op)
+  op.sig = uInt8ArrayToB64(
+    Object.values(nacl.sign.detached(strToUint8Array(message), sk))
+  );
+
+  try {
+    const response = await axios.post(BRIGHTID_NODE_URL + '/operations', op)
+    console.log(response)
+  } catch(error) {
+    console.log(JSON.stringify(error.response.data))
+  }
+  
+}
+
+const sendDollar = async (address) => {
+  const txConfig = {
+    to: address,
+    value: web3.util.toHex(web3.util.toWei(1)), 
+    gas: 21000,
+    gasPrice: 100000000,
+    nonce: await web3.eth.getTransactionCount(web3.eth.defaultAccount),
+    chainID: 1337 // ganache chain id
```

Comment:
We'll probably need a way to set the real chainID for IDChain (`0x4a` or `74`) and connect to an rpc node (https://idchain.one/rpc)  See "[add IDChain to metamask](https://brightid.gitbook.io/brightid/idchain/add-idchain-to-metamask)"